lib/libzrepl/data_conn.c

```diff
@@ -348,6 +356,8 @@ uzfs_zvol_worker(void *arg)
 
 		case ZVOL_OPCODE_SYNC:
 			uzfs_flush_data(zinfo->main_zv);
+			if (zinfo->clone_zv)
```

Comment:
lets check !ZVOL_IS_HEALTHY(zinfo->main_zv) here also instead of checking for zinfo->clone_zv).

Reply:
done.